### PR TITLE
Feature/User validation

### LINKED ISSUE #16: Add Class Validation to User Entity
Description:
Ensure that the data for the User entity is validated before it is processed. This involves adding class-validator decorators to the DTOs used for creating and updating users.

Tasks:

- [ ] Create DTOs for creating 
- [ ] and updating users.
- [ ] Add appropriate validation decorators to each field in the DTOs. Example: `@IsString(), @IsNotEmpty(), @IsEmail(), @IsOptional()`
- [ ] Update user controller methods to use these DTOs.

Acceptance Criteria:

- DTOs with validation decorators are created for user creation and update.
- User controller methods use these DTOs.
- Validation errors are properly handled and returned to the client.

### PR DESCRIPTION
**Pull request by GSzabi11**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,33 +1,76 @@
 {
+
```
Comment (csiszaralex) [inline package.json:2]: Why are you added new lines?

**Review by csiszaralex — CHANGES_REQUESTED** [review]
You have the same problems like #19 please check it and resolve the problems

File: apps/backend/src/user/entities/user.entity.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  email: string;
+
+  @IsString()
+  @IsOptional()
+  profilePicture: string;
+
+  @IsInt()
+  @IsOptional()
+  @Min(0)
+  groups: number;
```
Comment (balintking) [inline apps/backend/src/user/entities/user.entity.ts:27]: We don't need the groups and events list fields. They only exist in prisma's interpretation of the database, but they are not real.

File: apps/backend/src/user/dto/create-user.dto.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import { OmitType } from '@nestjs/swagger';
+
+import { User } from '../entities/user.entity';
+
+export class CreateUserDto extends OmitType(User, ['id', 'events', 'groups']) {}
```
Comment (balintking) [inline apps/backend/src/user/dto/create-user.dto.ts:5]: So we won't need them here either

**Review by balint-kiraly — CHANGES_REQUESTED** [review]
Sorry for the little late, please resolve these few things quickly and we can close this issue

**Review by balint-kiraly — APPROVED** [review]

**Review by csiszaralex — APPROVED** [review]
Nice work!

**Pull request merged**